app/Services/Model/Imp/ScheduleService.php

```diff
@@ -0,0 +1,262 @@
+            }
+            $schedule_event = $this->updateLocation($payload, $summit, $schedule_event);
+            $schedule_event = $this->updateEventDates($payload, $summit, $schedule_event);
+            $this->validateBlackOutTimesAndTimes($schedule_event);
+            $schedule->addScheduledSummitEvent($schedule_event);
+            $schedule_event->setUpdatedBy($member);
+            return $schedule_event;
+        });
+    }
+
+    /**
+     *@inheritDoc
+     */
+    public function unPublishProposedActivity(string $source, int $presentation_id):SummitProposedScheduleSummitEvent {
+        return $this->tx_service->transaction(function () use ($source, $presentation_id) {
+
+            $event = $this->event_repository->getById($presentation_id);
+
+            if (!$event instanceof SummitEvent)
+                throw new EntityNotFoundException(sprintf("event id %s does not exists!", $presentation_id));
+
+            $selection_plan = $event->getSelectionPlan();
+            if (!$selection_plan instanceof SelectionPlan)
+                throw new EntityNotFoundException("event id {$presentation_id} does not have a selection plan");
+
+            if (!$selection_plan->isAllowProposedSchedules())
+                throw new ValidationException("selection plan id {$selection_plan->getId()} does not allow proposed schedules");
+
+            $member = ResourceServerContext::getCurrentUser(false);
+
+            if (!$this->isAuthorizedUser( $member, $event->getSummit(), $event->getCategory()))
+                throw new AuthzException("User is not authorized to perform this action");
+
+            $schedules = $this->schedule_repository->getBySourceAndSummitId($source, $event->getSummitId());
+
+            if (count($schedules) == 0)
+                throw new EntityNotFoundException("schedule with source {$source} does not exists!");
+
+            $schedule = $schedules[0];
+
+            $schedule_event = $schedule->getScheduledSummitEventByEvent($event);
+
+            if (is_null($schedule_event))
+                throw new EntityNotFoundException(sprintf("schedule event for event id %s does not exists!", $presentation_id));
+
+            $schedule->removeScheduledSummitEvent($schedule_event);
+
+            return $schedule_event;
+        });
+    }
+
+    /**
+     *@inheritDoc
+     */
+    public function publishAll(string $source, int $summit_id, array $payload):SummitProposedSchedule
+    {
+        return $this->tx_service->transaction(function () use ($source, $summit_id, $payload) {
+
+            $schedules = $this->schedule_repository->getBySourceAndSummitId($source, $summit_id);
+
+            if (count($schedules) == 0) {
+                throw new EntityNotFoundException("schedule with source {$source} does not exists!");
+            }
+
+            $schedule = $schedules[0];
+
+            $filtered_schedule_events = [];
+
+            if (isset($payload['presentation_ids'])) {
```

Comment:
do this outside of the main transaction to remove lock contention

Reply:
done